include/deal.II/distributed/shared_tria.h

```diff
@@ -112,12 +144,42 @@ namespace parallel
          *     {
          *       parallel::shared::Triangulation<dim> tria(...,
          *                                                 parallel::shared::Triangulation<dim>::Settings::partition_custom_signal);
-         *       tria.signals.post_refinement.connect (std::bind(&mypartition<dim>, std::ref(shared_tria)));
+         *       tria.signals.post_refinement.connect (std::bind(&mypartition<dim>, std::ref(tria)));
+         *     }
+         *  @endcode
+         *
+         * An equivalent code using C++11 lambda functions would look like this:
+         *  @code
+         *     template <int dim>
+         *     void mypartition(parallel::shared::Triangulation<dim> &tria)
+         *     {
+         *       // user defined partitioning scheme: assign subdomain_ids
+         *       // round-robin in a mostly random way:
+         *       std::vector<unsigned int> assignment = {0,0,1,2,0,0,2,1,0,2,2,1,2,2,0,0};
+         *       typename Triangulation<dim>::active_cell_iterator
+         *         cell = tria.begin_active(),
+         *         endc = tria.end();
+         *       unsigned int index = 0;
+         *       for (; cell != endc; ++cell, ++index)
+         *         cell->set_subdomain_id(assignment[index%16]);
+         *     }
+         *
+         *     int main ()
+         *     {
+         *       parallel::shared::Triangulation<dim> tria(...,
+         *                                                 parallel::shared::Triangulation<dim>::Settings::partition_custom_signal);
+         *       tria.signals.post_refinement.connect ([&tria]()
+         *                                             {
+         *                                               // user defined partitioning scheme
```

Comment:
I don't understand this code example. Wouldn't you want to move the code from ``mypartition`` inside the lambda?